pkgs/flox-activation-scripts/default.nix

```diff
@@ -1,84 +1,106 @@
 {
   bash,
   coreutils,
+  daemonize,
+  fd,
   findutils,
+  flox-activations,
   getopt,
   gnused,
-  util-linux,
-  ld-floxlib,
-  runCommand,
-  shellcheck,
-  stdenv,
-  process-compose,
-  jq,
   iconv,
+  jq,
+  ld-floxlib,
   nawk,
-  fd,
-  flox-activations,
+  process-compose,
+  runCommandNoCC,
+  shellcheck,
   shfmt,
-  daemonize,
+  stdenv,
+  substituteAllFiles,
+  util-linux,
 }:
 let
-  ld-floxlib_so = if stdenv.isLinux then "${ld-floxlib}/lib/ld-floxlib.so" else "__LINUX_ONLY__";
-  ldconfig = if stdenv.isLinux then "${iconv}/bin/ldconfig" else "__LINUX_ONLY__";
-  # Some versions of Nix don't support `.` in name
+  activation-scripts = substituteAllFiles {
+    src = ../../assets/activation-scripts;
+    files = [ "." ]; # Perform recursive substitution on all files.
+    # Substitute all of the following variables.
+    inherit
+      bash
+      coreutils
+      daemonize
+      fd
+      findutils
+      getopt
+      gnused
+      jq
+      nawk
+      ;
+    # Note that substitution doesn't work with variables containing "-"
+    # so we need to create and use alternative names.
+    process_compose = process-compose;
+    flox_activations = flox-activations;
+    util_linux = util-linux;
+    # Make clear when packages are not available on Darwin.
+    ld_floxlib = if stdenv.isLinux then ld-floxlib else "__LINUX_ONLY__";
+    iconv = if stdenv.isLinux then iconv else "__LINUX_ONLY__";
+  };
+
+  # Create editorconfig for use in `shfmt` check. Note that some versions
+  # of Nix don't support `.` in name.
   editorconfig = builtins.path {
     name = "editorconfig";
     path = ../../.editorconfig;
   };
+
 in
-runCommand "flox-activation-scripts"
+runCommandNoCC "flox-activation-scripts"
   {
-    buildInputs = [
-      bash
-      coreutils
-      gnused
-      daemonize
+    nativeBuildInputs = [ gnused ];
+    outputs = [
+      "out"
+      "wrapper"
```

Comment:
Sure, except Nix package outputs cannot have "-" characters in them (because their names need to be defined as env variables as the build is run). How about `buildWrapper`?

Reply:
`out` and `wrapper` are not a great combination of names
I wonder what would be a better name for `$out` too 🤔

Reply:
SGTM 👍🏻

Reply:
Or `build_wrapper`? I see examples of both snake_case and camelCase in nixpkgs, but slightly more snake_case. Help me choose!!!

Reply:
I too wondered about a better name for `$out` but was wanting to limit the scope of this PR, and we're already consuming this as `$out` elsewhere. So if it's OK with everyone we can rename that in a different PR?

Updating patch to call it `builder_wrapper` (it's not used by anything yet .. #2343 will be the first PR that uses it)